syntax = "proto3";

option java_package = "net.yairsovran.scalica.processor";

package processor;

service Processor {
  // Do something with a newly-received Post from a user.
  // For now, just returns the SHA256 digest of the post's text
  // 
  rpc ProcessPost (ProcessPostRequest) returns (ProcessPostReply) {}
}

message ProcessPostRequest {
  int64 user_id = 1;  // User posting.
  int64 post_id = 2;  // The ID of the post.
  string text = 3;    // The text of the post.
}

message ProcessPostReply {
  string digest = 1;  // The sha256 digest of request.text.
  string augmented_text = 2;  // The processed text.
}
